### PR TITLE
Added call to fort-details to make spin pokestop more realistic

### PR DESCRIPTION
**Pull request by krosenvold**
The application appears to be pretty adamant about calling fort_details before spinning the pokestop.

## Motivation and Context
Make the application more like the real client

## How Has This Been Tested?
Code compiles :)

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pogom/account.py [hunk 1]
```diff
@@ -344,6 +346,20 @@ def spin_pokestop_request(api, fort, step_location):
         return False
 
 
+def fort_details_request(api, fort):
+    try:
+        req = api.create_request()
+        req.fort_details(
```
Comment (Alderon86) [inline pogom/account.py]: fort_details_response = req.fort_details(...

Comment (krosenvold) [inline pogom/account.py]: What is the reason for this double assignmentto the same variable ? I see this being done in several places

Comment (krosenvold) [inline pogom/account.py]: Yes, but I assume you're commenting about the assignment of req.fort_details() to fort_details_response, which is reassigned on the subsequent line ?

Comment (Alderon86) [inline pogom/account.py]: correct.

Comment (krosenvold) [inline pogom/account.py]: i've studied three other implementations of this logic an believe the current PR is correct, your additions would make it incorrect. Please show me some references if you believe  you Are right

File: pogom/account.py [hunk 2]
```diff
@@ -340,7 +342,21 @@ def spin_pokestop_request(api, fort, step_location):
...
+
+def fort_details_request(api, fort):
+    try:
+        req = api.create_request()
+        req.fort_details(
+            fort_id=fort['id'],
+            latitude=fort['latitude'],
+            longitude=fort['longitude'])
+        fort_details_response = req.call()
+        return fort_details_response
```
Comment (Alderon86) [inline pogom/account.py]: can we change to `response` instead of `fort_details_response`? would fit the actual code a bit more

**Comment by friscoMad** [comment]
@Krosenvold this needs a lot of changes to work with the latest api and RM changes, the fort_details_request should be almost the same as current gym_request just changing the main request method.

File: pogom/account.py [hunk 3]
```diff
@@ -589,6 +592,20 @@ def spin_pokestop_request(api, account, fort, step_location):
...
 
 
+def fort_details_request(api, fort):
+    try:
+        req = api.create_request()
+        req.fort_details(
+            fort_id=fort['id'],
+            latitude=fort['latitude'],
+            longitude=fort['longitude'])
+        response = req.call()
```
Comment (Alderon86) [inline pogom/account.py]:
```
        req = api.create_request()
        req.fort_details(
            fort_id=fort['id'],
            latitude=fort['latitude'],
            longitude=fort['longitude'])
               req.check_challenge()
        req.get_hatched_eggs()
        req.get_inventory()
        req.check_awarded_badges()
        req.get_buddy_walked(
        response = req.call()
```

File: pogom/account.py [hunk 4]
```diff
@@ -534,9 +534,12 @@ def spin_pokestop(api, account, fort, step_location):
     if in_radius((fort['latitude'], fort['longitude']), step_location,
                  spinning_radius):
         log.debug('Attempt to spin Pokestop (ID %s)', fort['id'])
-        time.sleep(random.uniform(0.8, 1.8))  # Do not let Niantic throttle
+
+        time.sleep(random.uniform(0.8, 1.8))
+        fort_details_request(api, fort)
+        time.sleep(random.uniform(0.8, 1.8))  # Don't let Niantic throttle
```
Comment (Arengo) [inline pogom/account.py]: ``.`` Kappa.

File: pogom/account.py (hunk 3 above)
Comment (Arengo) [inline pogom/account.py]: How about ``req.get_inbox(history=True)``? ;-)

File: pogom/account.py (hunk 1 above)
Comment (Arengo) [inline pogom/account.py]: Outdated.

File: pogom/account.py [hunk 5]
```diff
@@ -589,6 +592,20 @@ def spin_pokestop_request(api, account, fort, step_location):
         return False
 
 
+def fort_details_request(api, fort):
```
Comment (Arengo) [inline pogom/account.py]: According to recent request code changes, please consider adding in parsing of get_inventory with passed account argument.

Comment (krosenvold) [inline pogom/account.py]: I am not sure I understand this request, please clarify

File: pogom/account.py [hunk 6]
```diff
@@ -589,6 +592,26 @@ def spin_pokestop_request(api, account, fort, step_location):
...
+            latitude=fort['latitude'],
+            longitude=fort['longitude'])
+        req.check_challenge()
+        req.get_hatched_eggs()
+        req.get_inventory(last_timestamp_ms=account['last_timestamp_ms'])
+        req.check_awarded_badges()
+        req.get_buddy_walked()
+        req.get_inbox(is_history=True)
+        response = req.call()
+        return response
```
Comment (Arengo) [inline pogom/account.py]: ``parse_new_timestamp_ms(account, response)``

**Comment by sebastienvercammen** [comment]
@Arengo You approved, but you have two remaining comments:

> How about req.get_inbox(history=True)?

and

> parse_new_timestamp_ms(account, response)

Can you clarify whether you want to approve or if you want changes?

**Comment by krosenvold** [comment]
Both get_inbox and timestamp were fixed in the latest commits

**Pull request merged**